File: eAutoShop/eAutoShop.Model/Request/ReportRequest.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace eAutoShop.Model.Request
{
    public class ReportRequest
    {
        public string? Username { get; set; }
        public string? Role { get; set; }

        public DateTime? StartDate { get; set; }
        public DateTime? EndDate { get; set; }
    }
}

File: eAutoShop/eAutoShop.UI/eautoshop_desktop/lib/models/report/report_request.dart
class ReportRequest {
  final String? username;
  final String? role;
  final DateTime? startDate;
  final DateTime? endDate;

  const ReportRequest({this.username, this.role, this.startDate, this.endDate});

  Map<String, dynamic> toJson() {
    return {
      'username': username,
      'role': role,
      'startDate': startDate?.toIso8601String(),
      'endDate': endDate?.toIso8601String(),
    };
  }
}
